File: code/jvm/src/main/kotlin/com/gdeat/http/controllers/diagrams/models/getDiagram/GetDiagramOutputModel.kt
package com.gdeat.http.controllers.diagrams.models.getDiagram

import com.gdeat.service.diagrams.dtos.getDiagram.GetDiagramOutputDTO

class GetDiagramOutputModel(
    val Entities: Map<String, List<String>>,
    val Relations: Map<String, Map<String, String>>
) {
    constructor(getDiagramOutputDTO: GetDiagramOutputDTO) : this(
        Entities = getDiagramOutputDTO.Entities,
        Relations = getDiagramOutputDTO.Relations
    )
}

File: code/js/src/Services/services/diagrams/models/GetDiagramOutputModel.ts
import {SirenEntity} from "../../../media/siren/SirenEntity";

export interface GetDiagramOutputModel {
    entities: { [key: string]: string[] };
    relations: { [key: string]: { [key: string]: string } };
}

export type GetDiagramOutput = SirenEntity<GetDiagramOutputModel>